namelayer-spigot/src/main/java/vg/civcraft/mc/namelayer/database/GroupManagerDao.java

```diff
@@ -871,54 +970,29 @@ public void deleteGroup(String groupName){
 		}
 	}
 	
-	public void addMemberAsync(final UUID member, final String faction, final PlayerType role){
+	public void addMemberAsync(final UUID member, final Group group, final PlayerType role){
 		plugin.getServer().getScheduler().runTaskAsynchronously(plugin, new Runnable(){
 
 			@Override
 			public void run() {
-				addMember(member,faction,role);
+				addMember(member,group,role);
 			}
 			
 		});
 	}
 	
-	public void addMember(UUID member, String faction, PlayerType role){
+	public void addMember(UUID member, Group group, PlayerType role){
+		//TODO Redo this
 		try (Connection connection = db.getConnection();
 				PreparedStatement addMember = connection.prepareStatement(this.addMember)){
-			addMember.setString(1, member.toString());
-			addMember.setString(2, role.name());
-			addMember.setString(3, faction);
-			addMember.executeUpdate();
+		
 		} catch (SQLException e) {
 			logger.log(Level.WARNING, "Problem adding " + member + " as " + role.toString() 
-					+ " to group " + faction, e);
+					+ " to group " + group.getName(), e);
 		}			
 	}
 	
-	public List<UUID> getAllMembers(String groupName, PlayerType role){
-		List<UUID> members = new ArrayList<UUID>();
-		try (Connection connection = db.getConnection();
-				PreparedStatement getMembers = connection.prepareStatement(this.getMembers)){
-			getMembers.setString(1, groupName);
-			getMembers.setString(2, role.name());
-			try (ResultSet set = getMembers.executeQuery();) {
-				while(set.next()){
-					String uuid = set.getString(1);
-					if (uuid == null) {
-						continue;
-					}
-					members.add(UUID.fromString(uuid));
-				}
-			} catch (SQLException e) {
-				logger.log(Level.WARNING, "Problem getting all " + role.toString() + " for group " + groupName, e);
-			}
-		} catch (SQLException e) {
-			logger.log(Level.WARNING, "Problem preparing to get all " + role.toString() + " for group " + groupName, e);
-		}
-		return members;
-	}
-	
-	public void removeMemberAsync(final UUID member, final String group){
+	public void removeMemberAsync(final UUID member, final Group group){
```

Comment:
Why would you remove getAllMembers? For cache-filling this or something like it is critical.